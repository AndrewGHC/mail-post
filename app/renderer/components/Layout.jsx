import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Navbar from './navbar';
import {
  COLOUR_PRIMARY,
  COLOUR_SECONDARY,
  COLOUR_WHITE,
} from '../constants';

const HEADER_HEIGHT = '10vh';
const NAV_WIDTH = '20vw';

const Wrapper = styled.section`
  height: 100vh;
  max-height: 100vh;
  width: 100vw;
  max-width: 100vw;
  display: grid;
  grid-template:
      "hd hd" minmax(min-content, ${HEADER_HEIGHT})
      "nv mn" 1fr / minmax(min-content, ${NAV_WIDTH}) 1fr;
`;
const Header = styled.header`
  background-color: ${COLOUR_PRIMARY};
  grid-area: hd;
`;
const HeaderTitle = styled.div`
  color: ${COLOUR_WHITE};
  height: 100%;
  width: ${NAV_WIDTH};
  display: flex;
  align-items: center;
  margin-left: 16px;
`;
const Nav = styled.nav`
  background-color: ${COLOUR_SECONDARY};
  grid-area: nv;
`;
const Main = styled.main`
  background-color: ${COLOUR_WHITE};
  grid-area: mn;
`;

const Layout = ({ children }) => (
  <Wrapper>
    <Header>
      <HeaderTitle>
        <h1>
          MailPost
        </h1>
      </HeaderTitle>
    </Header>

    <Nav>
      <Navbar />
    </Nav>

    <Main>
      {children}
    </Main>
  </Wrapper>
);

Layout.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Layout;
