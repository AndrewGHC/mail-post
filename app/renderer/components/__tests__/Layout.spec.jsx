import React from 'react';
import { shallow } from 'enzyme';

import Layout from '../Layout';

describe('Layout', () => {
  it('renders children in the main element', async () => {
    const TestComponent = () => <div>hello</div>;
    const wrapper = shallow(
      <Layout>
        <TestComponent />
      </Layout>,
    );

    expect(wrapper.find('Layout__Main').contains(<TestComponent />)).toBe(true);
  });

  it('matches snapshot', async () => {
    const wrapper = shallow(<Layout />);
    expect(wrapper).toMatchSnapshot();
  });
});
