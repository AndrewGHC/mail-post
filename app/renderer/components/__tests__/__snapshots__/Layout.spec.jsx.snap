// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout matches snapshot 1`] = `
<Layout__Wrapper>
  <Layout__Header>
    <Layout__HeaderTitle>
      <h1>
        MailPost
      </h1>
    </Layout__HeaderTitle>
  </Layout__Header>
  <Layout__Nav>
    <Navbar />
  </Layout__Nav>
  <Layout__Main>
    <TestComponent />
  </Layout__Main>
</Layout__Wrapper>
`;
